charts/library/common/templates/classes/_service.tpl

```diff
@@ -93,6 +93,28 @@ spec:
   {{- end }}
   ports:
   {{- range $name, $port := $enabledPorts }}
+    {{- if $port.portRange }}
+      {{- range $portnum := untilStep $port.portRange.start $port.portRange.stop 1 }}
+    - port: {{ $portnum }}
+      targetPort: {{ $port.targetPort | default $port.port }}
+        {{- if $port.protocol }}
+          {{- if or ( eq $port.protocol "HTTP" ) ( eq $port.protocol "HTTPS" ) ( eq $port.protocol "TCP" ) }}
+      protocol: TCP
+          {{- else }}
+      protocol: {{ $port.protocol }}
+          {{- end }}
+        {{- else }}
+      protocol: TCP
+        {{- end }}
+      name: {{ $name }}
```

Comment:
Does `name` need to be unique? Could do `range-{{ $port.port }}` or similar if so